sklearn/bench.py

```diff
@@ -298,8 +318,124 @@ def logverbose(msg, verbose):
         print('@', msg)
 
 
+def convert_to_numpy1d(data):
+    if 'cudf' in str(type(data)):
+        data = data.to_pandas().values
+    if 'pandas' in str(type(data)):
+        data = data.values
+    return data.reshape((data.shape[0],))
+
+
 def accuracy_score(y, yp):
+    y = convert_to_numpy1d(y)
+    yp = convert_to_numpy1d(yp)
     return np.mean(y == yp)
 
 
-sklearn_set_no_input_check()
+def rmse_score(y, yp):
+    y = convert_to_numpy1d(y)
+    yp = convert_to_numpy1d(yp)
+    return float(np.sqrt(np.mean((y - yp)**2)))
+
+
+def convert_data(data, dtype, data_order, data_format):
+    if data_order == 'F':
+        data = np.asfortranarray(data, dtype)
+    elif data_order == 'C':
+        data = np.ascontiguousarray(data, dtype)
+
+    if data_format == 'numpy':
+        return data
+    elif data_format == 'pandas':
+        import pandas as pd
+
+        if len(data.shape) == 1:
+            return pd.Series(data)
+        else:
+            return pd.DataFrame(data)
+    elif data_format == 'cudf':
+        import cudf
+        import pandas as pd
+
+        return cudf.DataFrame.from_pandas(pd.DataFrame(data))
+
+
+def get_dtype(data):
+    if hasattr(data, 'dtype'):
+        return data.dtype
+    elif hasattr(data, 'values'):
+        return data.values.dtype
+    elif hasattr(data, 'dtypes'):
+        return str(data.dtypes[0])
+
+
+def load_data(params, generated_data=[], add_dtype=False, label_2d=False):
+    full_data = {
+        file: None for file in ['X_train', 'X_test', 'y_train', 'y_test']
+    }
+    param_vars = vars(params)
+    for element in full_data:
+        file_arg = f'file_{element}'
+        # load data from npy file if path is specified
+        if param_vars[file_arg] is not None:
+            full_data[element] = convert_data(
+                np.load(param_vars[file_arg].name),
+                params.dtype,
+                params.data_order,
+                params.data_format
+            )
+        # generate data if it's marked and path isn't specified
+        if full_data[element] is None and element in generated_data:
+            full_data[element] = convert_data(np.random.rand(*params.shape),
+                                              params.dtype, params.data_order,
+                                              params.data_format)
+        if full_data[element] is not None and 'y' in element and label_2d and hasattr(full_data[element], 'reshape'):
```

Comment:
Would be useful to have comments before each one of these if-blocks like you did in line 387.

Reply:
Comments were added.